polynote-frontend/polynote/ui/component/notebook/notebookconfig.ts

```diff
@@ -43,8 +46,10 @@ export class NotebookConfigEl extends Disposable {
         const exclusions = new Exclusions(configState.view("exclusions"), stateHandler);
         const resolvers = new Resolvers(configState.view("repositories"), stateHandler);
         const serverTemplatesHandler = ServerStateHandler.view("sparkTemplates").disposeWith(configState);
-        const spark = new SparkConf(configState.view("sparkConfig"), configState.view("sparkTemplate"), serverTemplatesHandler, stateHandler);
-        const kernel = new KernelConf(configState, stateHandler);
+        // we need a new state handler in order to avoid updating the config when the user changes the dropdown but doesn't hit 'Save'
```

Comment:
This seems like a bit of a smell to me. Would we still need this if we moved Scala version into SparkConf? (Or moved SparkConf into KernelConf)?

Reply:
Moved Scala version dropdown into the [Spark config panel](https://github.com/polynote/polynote/pull/1432/files#diff-541ec4938d59232cfd3df8fbe82a511cf67ebb63889005706d6f68499067cb02R446)